web/signature_manager.js

```diff
@@ -0,0 +1,680 @@
+      this.#drawPath.setAttribute("d", this.#drawPathString);
+
+      const finishDrawAC = new AbortController();
+      const listenerDrawOptions = {
+        signal: AbortSignal.any([signal, finishDrawAC.signal]),
+      };
+      this.#drawSVG.addEventListener(
+        "contextmenu",
+        noContextMenu,
+        listenerDrawOptions
+      );
+      this.#drawSVG.addEventListener(
+        "pointermove",
+        evt => {
+          evt.preventDefault();
+          let { offsetX: x, offsetY: y } = evt;
+          x = Math.round(x);
+          y = Math.round(y);
+          const drawPoints = this.#drawPoints;
+          if (
+            x < 0 ||
+            y < 0 ||
+            x > drawWidth ||
+            y > drawHeight ||
+            (x === drawPoints.at(-2) && y === drawPoints.at(-1))
+          ) {
+            return;
+          }
+          if (drawPoints.length >= 4) {
+            const [x1, y1, x2, y2] = drawPoints.slice(-4);
+            this.#drawPathString += `C${(x1 + 5 * x2) / 6} ${(y1 + 5 * y2) / 6} ${(5 * x2 + x) / 6} ${(5 * y2 + y) / 6} ${(x2 + x) / 2} ${(y2 + y) / 2}`;
+          } else {
+            this.#drawPathString += `L${x} ${y}`;
+          }
+          drawPoints.push(x, y);
+          this.#drawPath.setAttribute("d", this.#drawPathString);
+        },
+        listenerDrawOptions
+      );
+      this.#drawSVG.addEventListener(
+        "pointerup",
+        evt => {
+          const { pointerId: pId } = evt;
+          if (!isNaN(currentPointerId) && currentPointerId !== pId) {
+            return;
+          }
+          currentPointerId = NaN;
+          evt.preventDefault();
+          this.#drawSVG.releasePointerCapture(pId);
+          finishDrawAC.abort();
+          if (this.#drawPoints.length === 2) {
+            this.#drawPathString += `L${this.#drawPoints[0]} ${this.#drawPoints[1]}`;
+            this.#drawPath.setAttribute("d", this.#drawPathString);
+          }
+        },
+        listenerDrawOptions
+      );
+    };
+    if (this.#drawCurves) {
+      this.#drawSVG.addEventListener("pointerdown", drawCallback, options);
+    } else {
+      this.#drawPlaceholder.addEventListener(
+        "pointerdown",
+        drawCallback,
+        options
+      );
+    }
+    this.#drawThickness.addEventListener(
+      "input",
+      () => {
+        const { value: thickness } = this.#drawThickness;
+        this.#drawThickness.setAttribute(
+          "data-l10n-args",
+          JSON.stringify({ thickness })
+        );
+        if (!this.#drawCurves) {
+          return;
+        }
+        this.#drawPath.setAttribute("stroke-width", thickness);
+        this.#drawCurves.thickness = thickness;
+      },
+      options
+    );
+  }
+
+  #initImageTab(reset) {
+    if (reset) {
+      this.#resetTab("image");
+    }
+
+    this.#disableButtons(this.#imagePath);
+
+    const { signal } = this.#currentTabAC;
+    const options = { passive: true, signal };
```

Comment:
Nit: We could reduce object-creation below by defining two separate ones here.
```suggestion
    const options = { signal },
      passiveOptions = { passive: true, signal };
```